internal/k8s/client_test.go

```diff
@@ -161,37 +161,81 @@ func TestGetGroup(t *testing.T) {
 	}
 }
 
+func TestUpsertTimeout(t *testing.T) {
+	f := newClientTestFixture(t)
+	postgres := MustParseYAMLFromString(t, testyaml.PostgresYAML)
+
+	f.runner.pauseForever = true
+
+	// we can't use a fake clock with context.Context, so we'll cheat a bit
+	// and just pass Upsert an already expired context.
+	var cancel context.CancelFunc
+	f.ctx, cancel = context.WithDeadline(f.ctx, time.Now().Add(-time.Hour))
+	defer cancel()
+
+	_, err := f.client.Upsert(f.ctx, postgres)
+
+	require.Error(t, err)
+	require.Equal(t, "Killed kubectl. Hit timeout of 15s.", err.Error())
+}
+
 type call struct {
 	argv  []string
 	stdin string
 }
 
 type fakeKubectlRunner struct {
-	stdout string
-	stderr string
-	err    error
+	pauseForever bool
+	stdout       string
+	stderr       string
+	err          error
 
 	calls []call
 }
 
+func (f *fakeKubectlRunner) waitForDeadline(ctx context.Context) {
+	// ok not actually forever
```

Comment:
this comment is cute but also doesn't actually make sense because this function doesn't reference `pauseForever` anywhere?